sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCasts.scala

```diff
@@ -48,6 +48,10 @@ object CollationTypeCasts extends TypeCoercionRule {
     case eltExpr: Elt =>
       eltExpr.withNewChildren(eltExpr.children.head +: collateToSingleType(eltExpr.children.tail))
 
+    case overlay: Overlay =>
+      overlay.withNewChildren(collateToSingleType(Seq(overlay.input, overlay.replace))
+        ++ Seq(overlay.pos, overlay.len))
```

Comment:
Interesting thought/question: I do not know what Scala does when it calls Seq, but you can do single element append with :+. This operator seems to only allocate memory for 1 element and then write it to that place, while ++ allocates memory for all and then call Array.copy(...) for elements that need to be added. @cloud-fan Do you maybe know something a bit more on this topic?